content/en/docs/contributing/style-guide.md

```diff
@@ -0,0 +1,73 @@
+---
+title: "Style Guide"
+linkTitle: "Style Guide"
+weight: 80
+type: "docs"
+---
+
+cert-manager, like most Go projects, delegates almost all stylistic choices to `gofmt`,
+with `goimports` on top for organizing imports. Broadly speaking, if you set your editor to run
+`goimports` when you save a file, your code will be stylistically correct.
+
+cert-manager generally also follows the Kubernetes
+[coding conventions](https://www.kubernetes.dev/docs/guide/coding-convention/) and the Google
+[Go code review comments](https://github.com/golang/go/wiki/CodeReviewComments).
+
+## Organizing Imports
+
+Imports should be organized into 3 blocks, with each block separated by two newlines:
+
+```go
+import (
+	"stdlib"
+
+	"external"
+
+	"internal"
+)
+```
+
+An example might be the following, taken from
+[`pkg/acme/accounts/client.go`](https://github.com/jetstack/cert-manager/blob/0c71fe7795858b96cabcddabf706d997cd2fba3f/pkg/acme/accounts/client.go):
+
+```go
+
+import (
+	"crypto/rsa"
+	"crypto/tls"
+	"net"
+	"net/http"
+	"time"
+
+	acmeapi "golang.org/x/crypto/acme"
+
+	acmecl "github.com/jetstack/cert-manager/pkg/acme/client"
+	acmeutil "github.com/jetstack/cert-manager/pkg/acme/util"
+	cmacme "github.com/jetstack/cert-manager/pkg/apis/acme/v1"
+	"github.com/jetstack/cert-manager/pkg/metrics"
+	"github.com/jetstack/cert-manager/pkg/util"
+)
+```
+
+Once this manual split of standard library, external and internal imports has been made, it will be
+enforced automatically by `goimports` when executed in the future.
+
+## Go Module Compatibility
+
+It's important to note that cert-manager `v1` doesn't generally provide any guarantees that it can be
+imported and used as a module by other programs, even for publicly exported functions, variables,
+constants, interfaces and types under `pkg/`.
+
+Only the following modules can be explicitly relied upon to maintain the go module compatibility
+guarantee in cert-manager `v1`:
+
+- `pkg/api`
+- `pkg/apis`
+- `pkg/acme/webhook`
+- `pkg/client`
+- `pkg/dns/util`
+- `pkg/issuer/acme/dns/util`
+- `pkg/util/pki`
```

Comment:
I don't think this should be on this list either - if there are specific functions we expect to be stable, can we separate them out? It's typically been quite messy in here 😬

Reply:
I added this because it's used in a bunch of existing open source stuff already: https://pkg.go.dev/github.com/jetstack/cert-manager@v1.3.1/pkg/util/pki?tab=importedby - as originally pointed out by @irbekrm

Reply:
I don't think that should be our benchmark for what should be stable though - I think that indicates we need to review the code in these packages to ensure it's all in a state where we _are_ comfortable supporting it long term, before doing so.